modules/Attendance/attendance.php

```diff
@@ -16,488 +16,447 @@
+                        }
+
+                        // determine how to display link and content
+                        // according to status
+                        switch ($day['status']) {
+                            case 'na':
+                                $class = 'highlightNoData';
+                                $content = __('NA');
+                                break;
+                            case 'present':
+                                $class = 'highlightPresent';
+                                $content = Format::link($link, $content);
+                                break;
+                            case 'absent':
+                                $class = 'highlightAbsent';
+                                $content = Format::link($link, $content);
+                                break;
+                            default:
+                                $class = 'highlightNoData';
+                                break;
+                        }
+
+                        $dayTable .= "<td class=\"{$class}\" style=\"padding: 12px !important;\">{$content}</td>";
+                    }
+                    $i += $j;
+                    $dayTable .= '</tr>';
+                }
+
+                $dayTable .= '</table>';
+                return $dayTable;
+            });
+        $dailyAttendanceTable->addColumn('today', __('Today'))
+            ->width('40px')
+            ->format(function ($row) use ($session) {
+                switch ($row['today']) {
+                    case 'taken':
+                        // attendance taken
+                        return '<img src="./themes/' . $session->get('gibbonThemeName') . '/img/iconTick.png"/>';
+                    case 'not taken':
+                        // attendance not taken
+                        return '<img src="./themes/' . $session->get('gibbonThemeName') . '/img/iconCross.png"/>';
+                    case 'not timetabled':
+                        // class not timetabled on the day
+                        return '<span title="'.__('This class is not timetabled to run on the specified date. Attendance may still be taken for this group however it currently falls outside the regular schedule for this class.').'">' .
+                            __('N/A') . '</span>';
+                }
+            });
+        $dailyAttendanceTable->addColumn('in', __('In'))
+            ->width('40px');
+        $dailyAttendanceTable->addColumn('out', __('Out'))
+            ->width('40px');
+
+        // action column, if user has the permission
+        if (isActionAccessible($guid, $connection2, $takeAttendanceURL)) {
+            $dailyAttendanceTable->addActionColumn()
+                ->width('50px')
+                ->addAction('takeAttendance')
+                ->setLabel(__('Take Attendance'))
+                ->setIcon('attendance')
+                ->setURL($takeAttendanceURL)
+                ->addParam($rowID)
```

Comment:
This is throwing an ArgumentCountError when I test it, looks like it's because `addActionColumn()` returns an ActionColumn object, and then in your chained methods, `addAction()` returns an Action. The `addParam()` without passing a value is a method on ActionColumn. Moving the addParam call above addAction fixes it. Splitting this chained call may help clarify what's going on too.